src/autopas/containers/verletListsCellBased/verletListsCells/neighborLists/VLCCellPairNeighborList.h

```diff
@@ -114,38 +133,90 @@ class VLCCellPairNeighborList : public VLCNeighborListInterface<Particle> {
       }
     }
 
-    applyBuildFunctor(linkedCells, useNewton3, cutoff, skin, interactionLength, buildTraversalOption);
+    applyBuildFunctor(linkedCells, useNewton3, cutoff, skin, interactionLength, buildTraversalOption, buildType);
+  }
+
+  /**
+   * @copydoc VLCNeighborListInterface::generateSoAFromAoS()
+   * */
+  void generateSoAFromAoS(LinkedCells<Particle> &linkedCells) override {
+    _soaNeighborList.clear();
+
+    std::unordered_map<Particle *, size_t> _particleToIndex;
+    _particleToIndex.reserve(linkedCells.getNumParticles());
+    size_t i = 0;
+    for (auto iter = linkedCells.begin(IteratorBehavior::haloOwnedAndDummy); iter.isValid(); ++iter, ++i) {
+      _particleToIndex[&(*iter)] = i;
+    }
+
+    auto &cells = linkedCells.getCells();
+    size_t cellsSize = cells.size();
+    _soaNeighborList.resize(cellsSize);
+
+    // iterate over aosNeighborList and insert indices instead of particles in the respective place in soaNeighborList
+    for (size_t firstCellIndex = 0; firstCellIndex < _aosNeighborList.size(); ++firstCellIndex) {
+      auto &aosCurrentCell = _aosNeighborList[firstCellIndex];
+      auto &soaCurrentCell = _soaNeighborList[firstCellIndex];
+      soaCurrentCell.resize(aosCurrentCell.size());
+
+      for (size_t secondCellIndex = 0; secondCellIndex < aosCurrentCell.size(); ++secondCellIndex) {
+        auto &aosCellPair = aosCurrentCell[secondCellIndex];
+        auto &soaCellPair = soaCurrentCell[secondCellIndex];
+        soaCellPair.reserve(aosCellPair.capacity());
+
+        size_t particleIndexInCurrentCell = 0;
+        for (auto &aosParticleAndParticleList : aosCellPair) {
+          Particle *currentParticle = aosParticleAndParticleList.first;
+          size_t currentParticleGlobalIndex = _particleToIndex.at(currentParticle);
+
+          soaCellPair.emplace_back(
+              std::make_pair(currentParticleGlobalIndex, std::vector<size_t, autopas::AlignedAllocator<size_t>>()));
+
+          for (auto &neighborOfCurrentParticle : aosParticleAndParticleList.second) {
+            soaCellPair[particleIndexInCurrentCell].second.emplace_back(_particleToIndex.at(neighborOfCurrentParticle));
+          }
+          particleIndexInCurrentCell++;
```

Comment:
this code is basically the same as for the `VLCAllCellsNeighborList`. We should strive to avoid code duplication so maybe we can outsource the common part as a free function to (something like) `VerletListsCellsHelpers`?

Reply:
-> postponed ad infinitum